File: Difficulty: Easy/Minimum distance in an Array/minimum-distance-in-an-array.py
class Solution:
    def minDist(self, arr, x, y):
        x_index = -1
        y_index = -1
        ans = float('inf')
        for i in range(len(arr)):
            if arr[i] == x:
                x_index = i
            elif arr[i] == y:
                y_index = i
            if x_index != -1 and y_index != -1:
                ans = min(ans, abs(x_index - y_index))
        if ans == float('inf'):
            return -1
        else:
            return ans

File: Difficulty: Easy/Minimum distance in an Array/minimum-distance-in-an-array.cpp
class Solution {
  public:
    int minDist(vector<int>& arr, int x, int y) {
        int n = arr.size();
        int lastPos = -1;   // last index where x or y was seen
        int minDist = INT_MAX;
        bool foundX = false, foundY = false;
        
        for (int i = 0; i < n; i++) {
            if (arr[i] == x || arr[i] == y) {
                if (lastPos != -1 && arr[i] != arr[lastPos]) {
                    minDist = min(minDist, i - lastPos);
                }
                lastPos = i;
                if (arr[i] == x) foundX = true;
                if (arr[i] == y) foundY = true;
            }
        }
        
        if (!foundX || !foundY) return -1;
        return (minDist == INT_MAX) ? -1 : minDist;
    }
};